pg_bm25/src/api/search.rs

```diff
@@ -1,3 +1,4 @@
+#![allow(unused)]
```

Comment:
Silly question but why do we have these in a lot of places?

Reply:
I'm glad you're pointing this out, because it's easy to leave this in after temporarily `allow`-ing things for dev... I just did a review of all the remaining ones and feel good about what's still in the codebase.

I took a strategy with this PR to create type mappings for _all_ of the queries available in Tantivy, and then created functions exposing them to Postgres. After some testing, I realized it was a little unclear how certain queries should work in Postgres, like:
`MoreLikeThis`
`Range`
`FastFieldRangeWeight`

Since the work to implement these functions is done, I figured I would leave them in for some time while I think about whether it makes sense to fully support them. I'll clean them up in a couple of weeks if I decide there's no use case, and then they'll at least be in our git history.

Reply:
Sounds good. Same thing with the `dead_code` ?